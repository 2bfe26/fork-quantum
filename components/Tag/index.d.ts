import { ReactNode, MouseEventHandler, FC, HTMLAttributes } from 'react';

export interface TagProps extends HTMLAttributes<HTMLInputElement> {
    bold?: boolean;
    inverted?: boolean;
    children?: ReactNode[] | ReactNode;
    onClose?: MouseEventHandler<HTMLButtonElement>;
    size?: 'small' | 'medium' | 'large';
    skin?: 'neutral' | 'primary' | 'success' | 'warning' | 'error';
    stroked?: boolean;
    text?: string;
    theme?: {
        colors?: {};
        components?: {
            tag?: {};
        };
        spacing?: {};
    };
}

declare const Tag: FC<TagProps>;
export default Tag;
