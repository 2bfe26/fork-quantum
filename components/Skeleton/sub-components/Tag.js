import styled from 'styled-components';
import { Rect } from './Rect';

export const Tag = styled(Rect)`
  border-radius: 8px;
  margin-right: 8px;

  height: ${({ size, theme: { baseFontSize, spacing } }) => {
    const sizes = {
      small: baseFontSize * 0.875,
      medium: baseFontSize * 1.125,
      large: baseFontSize * 1.25,
    };

    return `${(sizes[size] || sizes.medium) + spacing.xxsmall * 2}px`;
  }};
`;
