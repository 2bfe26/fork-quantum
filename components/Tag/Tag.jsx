import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Colors from '../Colors';
import Icon from '../Icon';

const skinFontColors = {
  inverted: Colors.WHITE,
  blue: Colors.BLUE[500],
  success: Colors.SUCCESS[900],
  warning: Colors.WARNING[900],
  error: Colors.ERROR[900],
  'blue-text': Colors.BLUE[500],
};

const skins = {
  default: `background-color: ${Colors.BLACK[100]};`,
  stroked: `
    border: 1px solid;
    background: none;
  `,
  inverted: `
    color: ${skinFontColors.inverted}
    background-color: ${Colors.BLACK[400]};
  `,
  blue: `
    color: ${skinFontColors.blue}
    background-color: ${Colors.BLUE[200]};
  `,
  success: `
    color: ${skinFontColors.success}
    background-color: ${Colors.SUCCESS[200]};
  `,
  warning: `
    color: ${skinFontColors.warning}
    background-color: ${Colors.WARNING[200]};
  `,
  error: `
    color: ${skinFontColors.error}
    background-color: ${Colors.ERROR[200]};
  `,
  'blue-text': `
    color: ${skinFontColors['blue-text']}
    background-color: ${Colors.BLACK[100]};
  `,
};

const sizes = {
  small: `
    font-size: 12px;
    padding-top: 3px;
    padding-bottom: 3px;
  `,
  large: `
    font-size: 18px;
  `,
};

const Wrapper = styled.div`
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  ${({ bold }) => bold && `font-weight: bold;`}
  ${({ skin }) => skins[skin]}
  ${({ size }) => sizes[size]}
`;

const Content = styled.div`
  ${({ closable }) =>
    closable &&
    `
    display: flex;
    align-items: center;
  `}
`;

const iconSizes = {
  small: '14px',
  medium: '18px',
  large: '20px',
};

const ResetedButton = styled.button`
  background: none;
  border: none;
  cursor: pointer;
  height: 18px;
  margin-left: 4px;
  padding: 0;
  ${({ size }) => `height: ${iconSizes[size]};`}
`;

const SmallIcon = styled(Icon)`
  ${({ size }) => `font-size: ${iconSizes[size]} !important;`}
`;

class Tag extends Component {
  constructor(props) {
    super(props);

    this.state = { hidden: false };

    this.hide = this.hide.bind(this);
  }

  hide() {
    this.setState({ hidden: true });
  }

  render() {
    const { hidden } = this.state;
    if (hidden) return '';

    const { children, text, closable, skin } = this.props;

    return (
      <Wrapper {...this.props}>
        <Content closable={closable}>
          {children || text}
          {closable && (
            <ResetedButton onClick={this.hide} {...this.props}>
              <SmallIcon
                {...this.props}
                name="close"
                skin={skinFontColors[skin]}
              />
            </ResetedButton>
          )}
        </Content>
      </Wrapper>
    );
  }
}

Tag.propTypes = {
  skin: PropTypes.oneOf([
    'default',
    'stroked',
    'inverted',
    'blue',
    'success',
    'warning',
    'error',
    'blue-text',
  ]),
  children: PropTypes.string,
  text: PropTypes.string,
  bold: PropTypes.bool,
  closable: PropTypes.bool,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
};

Tag.defaultProps = {
  skin: 'default',
  text: 'Tag text',
  bold: false,
  children: '',
  closable: false,
  size: 'medium',
};

export default Tag;
