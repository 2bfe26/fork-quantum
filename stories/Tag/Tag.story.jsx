import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  Tab,
  Example,
  AutoExample,
  Title,
} from '@catho-private/quantum-storybook-ui';
import { Tag } from '../../components';

const skinsCode = `<Tag>Default</Tag>
<Tag skin="stroked">Stroked</Tag>
<Tag skin="inverted">Inverted</Tag>
<Tag skin="blue">Blue</Tag>
<Tag skin="success">Success</Tag>
<Tag skin="warning">Warning</Tag>
<Tag skin="error">Error</Tag>
<Tag skin="blue-text">Blue Text</Tag>`;

const skinsExample = (
  <>
    <Title>Skins</Title>
    <Example
      component={
        <>
          <Tag>Default</Tag>
          <Tag skin="stroked">Stroked</Tag>
          <Tag skin="inverted">Inverted</Tag>
          <Tag skin="blue">Blue</Tag>
          <Tag skin="success">Success</Tag>
          <Tag skin="warning">Warning</Tag>
          <Tag skin="error">Error</Tag>
          <Tag skin="blue-text">Blue Text</Tag>
        </>
      }
      code={skinsCode}
    />
  </>
);

const closableCodes = `<Tag closable>Default</Tag>
<Tag closable skin="stroked">Stroked</Tag>
<Tag closable skin="inverted">Inverted</Tag>
<Tag closable skin="blue">Blue</Tag>
<Tag closable skin="success">Success</Tag>
<Tag closable skin="warning">Warning</Tag>
<Tag closable skin="error">Error</Tag>
<Tag closable skin="blue-text">Blue Text</Tag>`;

const closableExample = (
  <>
    <Title>Closables</Title>
    <Example
      component={
        <>
          <Tag closable>Default</Tag>
          <Tag closable skin="stroked">
            Stroked
          </Tag>
          <Tag closable skin="inverted">
            Inverted
          </Tag>
          <Tag closable skin="blue">
            Blue
          </Tag>
          <Tag closable skin="success">
            Success
          </Tag>
          <Tag closable skin="warning">
            Warning
          </Tag>
          <Tag closable skin="error">
            Error
          </Tag>
          <Tag closable skin="blue-text">
            Blue Text
          </Tag>
        </>
      }
      code={closableCodes}
    />
  </>
);

const sizesCode = `<Tag skin="small">Small</Tag>
<Tag>Medium (Default)</Tag>
<Tag skin="large">Large</Tag>`;

const sizesExample = (
  <>
    <Title>Sizes</Title>
    <Example
      component={
        <>
          <Tag size="small">Small</Tag>
          <Tag>Medium (Default)</Tag>
          <Tag size="large">Large</Tag>
        </>
      }
      code={sizesCode}
    />
  </>
);

const exampleTab = (
  <Tab title="Examples">
    {skinsExample}
    {closableExample}
    {sizesExample}
  </Tab>
);

storiesOf('16. Tag', module).add('Tag', () => (
  <AutoExample component={Tag} additionalTabs={exampleTab} />
));
