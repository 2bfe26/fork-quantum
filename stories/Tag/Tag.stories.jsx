import Tag from '../../components/Tag';

const Template = args => {
  const { children: Component, ...rest } = args;
  return <Tag {...rest}>{Component}</Tag>;
};

export const Default = Template.bind({});

export const Skin = Template.bind({});
Skin.args = {
  skin: 'success',
};

export const Inverted = Template.bind({});
Inverted.args = {
  inverted: true,
};

export const Stroked = Template.bind({});
Stroked.args = {
  stroked: true,
};
