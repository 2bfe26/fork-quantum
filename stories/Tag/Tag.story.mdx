import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';

import Tag from '../../components/Tag';

import { Header } from '../shared';

import { 
  Default,
  Skin,
  Inverted,
  Stroked,
  Closable
} from './Tag.stories.jsx';

<Meta title="Tag" component={Tag} />

<Header title="Tag">
  Tags are Used to make stylized markings.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import { Tag } from '@catho/quantum';
```

<Canvas>
  <Story story={Default} />
</Canvas>

## <a id="api"></a>API

<ArgsTable of={Tag} />

## <a id="examples"></a>Examples

### Skin

<Canvas>
  <Story story={Skin} />
</Canvas>

### Inverted

<Canvas>
  <Story story={Inverted} />
</Canvas>

### Stroked

<Canvas>
  <Story story={Stroked} />
</Canvas>


