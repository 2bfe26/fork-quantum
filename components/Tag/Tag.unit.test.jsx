import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Tag from './Tag';

const testSnapshot = props => {
  const propsList = Array.isArray(props) ? props : [props];

  propsList.forEach(p => {
    expect(renderer.create(<Tag {...p} />).toJSON()).toMatchSnapshot();
  });
};

describe('Tag component', () => {
  it('without props', () => testSnapshot());
  it('text prop / children', () =>
    testSnapshot([{ text: 'example' }, { children: 'example' }]));
  it('bold', () => testSnapshot({ bold: true }));
  it('stroked skin', () => testSnapshot({ skin: 'stroked' }));
  it('inverted skin', () => testSnapshot({ skin: 'inverted' }));
  it('blue skin', () => testSnapshot({ skin: 'blue' }));
  it('success skin', () => testSnapshot({ skin: 'success' }));
  it('warning skin', () => testSnapshot({ skin: 'warning' }));
  it('error skin', () => testSnapshot({ skin: 'error' }));
  it('blue-text skin', () => testSnapshot({ skin: 'blue-text' }));

  describe('Closable', () => {
    it('it is rendering well', () => {
      testSnapshot({ closable: true });
      testSnapshot({ closable: true, skin: 'stroked' });
      testSnapshot({ closable: true, skin: 'inverted' });
      testSnapshot({ closable: true, skin: 'blue' });
      testSnapshot({ closable: true, skin: 'success' });
      testSnapshot({ closable: true, skin: 'warning' });
      testSnapshot({ closable: true, skin: 'error' });
      testSnapshot({ closable: true, skin: 'blue-text' });
    });

    it('is removed when close button is clicked', () => {
      const tag = shallow(<Tag closable />);
      const button = tag.childAt(1);

      expect(tag.state('hidden')).toBe(false);

      button.simulate('click', {});

      expect(tag.state('hidden')).toBe(true);
      expect(tag.html()).toBeNull();
    });
  });
});
