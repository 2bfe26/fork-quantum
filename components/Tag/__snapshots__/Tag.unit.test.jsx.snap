// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> Closable is rendering well 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  background-color: #f2f2f2;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background: none;
  border: none;
  cursor: pointer;
  height: 18px;
  margin-left: 4px;
  padding: 0;
  height: 18px;
}

.c3 {
  font-size: 18px !important;
}

<div
  className="c0"
  size="medium"
>
  <div
    className="c1"
  >
    Tag text
    <button
      className="c2"
      onClick={[Function]}
      size="medium"
    >
      <span
        aria-hidden="true"
        bold={false}
        className="material-icons MuiIcon-root-1 c3"
        closable={true}
        size="medium"
        style={
          Object {
            "color": "",
          }
        }
        text="Tag text"
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`<Tag /> Closable is rendering well 2`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  border: 1px solid;
  background: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background: none;
  border: none;
  cursor: pointer;
  height: 18px;
  margin-left: 4px;
  padding: 0;
  height: 18px;
}

.c3 {
  font-size: 18px !important;
}

<div
  className="c0"
  size="medium"
>
  <div
    className="c1"
  >
    Tag text
    <button
      className="c2"
      onClick={[Function]}
      size="medium"
    >
      <span
        aria-hidden="true"
        bold={false}
        className="material-icons MuiIcon-root-1 c3"
        closable={true}
        size="medium"
        style={
          Object {
            "color": "",
          }
        }
        text="Tag text"
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`<Tag /> Closable is rendering well 3`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #FFFFFF;
  background-color: #999999;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background: none;
  border: none;
  cursor: pointer;
  height: 18px;
  margin-left: 4px;
  padding: 0;
  height: 18px;
}

.c3 {
  font-size: 18px !important;
}

<div
  className="c0"
  size="medium"
>
  <div
    className="c1"
  >
    Tag text
    <button
      className="c2"
      onClick={[Function]}
      size="medium"
    >
      <span
        aria-hidden="true"
        bold={false}
        className="material-icons MuiIcon-root-1 c3"
        closable={true}
        size="medium"
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
        text="Tag text"
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`<Tag /> Closable is rendering well 4`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #1355d0;
  background-color: #e5edfc;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background: none;
  border: none;
  cursor: pointer;
  height: 18px;
  margin-left: 4px;
  padding: 0;
  height: 18px;
}

.c3 {
  font-size: 18px !important;
}

<div
  className="c0"
  size="medium"
>
  <div
    className="c1"
  >
    Tag text
    <button
      className="c2"
      onClick={[Function]}
      size="medium"
    >
      <span
        aria-hidden="true"
        bold={false}
        className="material-icons MuiIcon-root-1 c3"
        closable={true}
        size="medium"
        style={
          Object {
            "color": "#1355d0",
          }
        }
        text="Tag text"
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`<Tag /> Closable is rendering well 5`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #3c6510;
  background-color: #edfadf;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background: none;
  border: none;
  cursor: pointer;
  height: 18px;
  margin-left: 4px;
  padding: 0;
  height: 18px;
}

.c3 {
  font-size: 18px !important;
}

<div
  className="c0"
  size="medium"
>
  <div
    className="c1"
  >
    Tag text
    <button
      className="c2"
      onClick={[Function]}
      size="medium"
    >
      <span
        aria-hidden="true"
        bold={false}
        className="material-icons MuiIcon-root-1 c3"
        closable={true}
        size="medium"
        style={
          Object {
            "color": "#3c6510",
          }
        }
        text="Tag text"
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`<Tag /> Closable is rendering well 6`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #a36300;
  background-color: #ffefd6;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background: none;
  border: none;
  cursor: pointer;
  height: 18px;
  margin-left: 4px;
  padding: 0;
  height: 18px;
}

.c3 {
  font-size: 18px !important;
}

<div
  className="c0"
  size="medium"
>
  <div
    className="c1"
  >
    Tag text
    <button
      className="c2"
      onClick={[Function]}
      size="medium"
    >
      <span
        aria-hidden="true"
        bold={false}
        className="material-icons MuiIcon-root-1 c3"
        closable={true}
        size="medium"
        style={
          Object {
            "color": "#a36300",
          }
        }
        text="Tag text"
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`<Tag /> Closable is rendering well 7`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #ac001a;
  background-color: #ffe5e9;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background: none;
  border: none;
  cursor: pointer;
  height: 18px;
  margin-left: 4px;
  padding: 0;
  height: 18px;
}

.c3 {
  font-size: 18px !important;
}

<div
  className="c0"
  size="medium"
>
  <div
    className="c1"
  >
    Tag text
    <button
      className="c2"
      onClick={[Function]}
      size="medium"
    >
      <span
        aria-hidden="true"
        bold={false}
        className="material-icons MuiIcon-root-1 c3"
        closable={true}
        size="medium"
        style={
          Object {
            "color": "#ac001a",
          }
        }
        text="Tag text"
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`<Tag /> Closable is rendering well 8`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #1355d0;
  background-color: #f2f2f2;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background: none;
  border: none;
  cursor: pointer;
  height: 18px;
  margin-left: 4px;
  padding: 0;
  height: 18px;
}

.c3 {
  font-size: 18px !important;
}

<div
  className="c0"
  size="medium"
>
  <div
    className="c1"
  >
    Tag text
    <button
      className="c2"
      onClick={[Function]}
      size="medium"
    >
      <span
        aria-hidden="true"
        bold={false}
        className="material-icons MuiIcon-root-1 c3"
        closable={true}
        size="medium"
        style={
          Object {
            "color": "#1355d0",
          }
        }
        text="Tag text"
      >
        close
      </span>
    </button>
  </div>
</div>
`;

exports[`<Tag /> Sizes large 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  background-color: #f2f2f2;
  font-size: 18px;
}

<div
  className="c0"
  size="large"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Sizes medium 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  background-color: #f2f2f2;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Sizes small 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  background-color: #f2f2f2;
  font-size: 12px;
  padding-top: 3px;
  padding-bottom: 3px;
}

<div
  className="c0"
  size="small"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Skins blue 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #1355d0;
  background-color: #e5edfc;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Skins blue-text 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #1355d0;
  background-color: #f2f2f2;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Skins default 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  background-color: #f2f2f2;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Skins error 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #ac001a;
  background-color: #ffe5e9;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Skins inverted 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #FFFFFF;
  background-color: #999999;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Skins stroked 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  border: 1px solid;
  background: none;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Skins success 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #3c6510;
  background-color: #edfadf;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> Skins warning 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  color: #a36300;
  background-color: #ffefd6;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> with bold prop 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  font-weight: bold;
  background-color: #f2f2f2;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;

exports[`<Tag /> with only text prop / children 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  background-color: #f2f2f2;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    example
  </div>
</div>
`;

exports[`<Tag /> with only text prop / children 2`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  background-color: #f2f2f2;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    example
  </div>
</div>
`;

exports[`<Tag /> without props 1`] = `
.c0 {
  border-radius: 8px;
  box-sizing: border-box;
  display: inline-block;
  line-height: 1.5;
  margin-right: 8px;
  padding: 4px 12px;
  background-color: #f2f2f2;
}

<div
  className="c0"
  size="medium"
>
  <div
    className=""
  >
    Tag text
  </div>
</div>
`;
